ui/raidboss/data/05-shb/eureka/zadnor.js

```diff
@@ -81,30 +133,1155 @@ export default {
+      // 873 = Left Face
+      // 874 = Right Face
+      netRegex: NetRegexes.gainsEffect({ source: '4th-Make Shemhazai', effectId: ['871', '872', '873', '874'] }),
+      condition: (data, matches) => data.ce === 'grave' && data.me === matches.target,
+      alertText: (_data, matches, output) => {
+        const effectId = matches.effectId.toUpperCase();
+        if (effectId === '871')
+          return output.forward();
+        if (effectId === '872')
+          return output.backward();
+        if (effectId === '873')
+          return output.left();
+        if (effectId === '874')
+          return output.right();
+      },
+      outputStrings: {
+        forward: {
+          en: 'March Forward Into Middle',
+        },
+        backward: {
+          en: 'March Backward Into Middle',
+        },
+        left: {
+          en: 'March Left Into Middle',
+        },
+        right: {
+          en: 'March Right Into Middle',
+        },
+      },
+    },
     // ***** With Diremite and Main *****
+    {
+      id: 'Zadnor Diremite Crystal Needle',
+      netRegex: NetRegexes.startsUsing({ source: 'Hedetet', id: '5E15' }),
+      condition: tankBusterOnParty('diremite'),
+      response: Responses.tankBuster(),
+    },
+    {
+      id: 'Zadnor Diremite Shardstrike',
+      netRegex: NetRegexes.startsUsing({ source: 'Hedetet', id: '5E17' }),
+      condition: (data, matches) => data.ce === 'diremite' && data.me === matches.target,
+      response: Responses.spread(),
+    },
+    {
+      id: 'Zadnor Diremite Hailfire You',
+      netRegex: NetRegexes.headMarker({ id: limitCutHeadmarkers }),
+      condition: (data, matches) => data.ce === 'diremite' && data.me === matches.target,
+      preRun: (data, matches) => {
+        data.diremiteHailfire = data.diremiteHailfire || [];
+        data.diremiteHailfire.push(matches.target);
+      },
+      alertText: (_data, matches, output) => {
+        const id = matches.id;
+        const num = limitCutHeadmarkers.indexOf(id) + 1;
+        if (num < 1)
+          return;
+        const numStr = output[`num${num}`]();
+        return output.text({ num: numStr });
+      },
+      outputStrings: {
+        num1: numberOutputStrings[1],
+        num2: numberOutputStrings[2],
+        num3: numberOutputStrings[3],
+        num4: numberOutputStrings[4],
+        text: {
+          en: '${num} (spread for laser)',
+        },
+      },
+    },
+    {
+      id: 'Zadnor Diremite Hailfire Not You',
+      netRegex: NetRegexes.headMarker({ id: limitCutHeadmarkers, capture: false }),
+      condition: (data) => data.ce === 'diremite',
+      delaySeconds: 0.5,
+      suppressSeconds: 1,
+      infoText: (data, _matches, output) => {
+        if (data.diremiteHailfire && !data.diremiteHailfire.includes(data.me))
+          return output.text();
+      },
+      run: (data) => delete data.diremiteHailfire,
+      outputStrings: {
+        text: {
+          en: 'Avoid Hailfire Lasers',
```

Comment:
Does 'Hailfire' add something to this callout?
```suggestion
          en: 'Get Towers',
          de: 'Türme nehmen',
          fr: 'Prenez les tours',
          ja: '塔を踏む',
          cn: '踩塔',
          ko: '장판 들어가기',
```